### PR TITLE
feat: config-files feature

### PR DESCRIPTION
**Pull request by luke-biel**
Change is big, had to split it

### CODE REVIEW

**Comment by luke-biel** [comment]
Some extra changes since 5 part split merge: https://github.com/epiphany-platform/CommonDataLayer/pull/470/commits/aea8b31144d9513a9b468ae3c1cd8f01ac6af34d. Rest was reviewed.

File: crates/api/src/main.rs [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
+async fn main() -> anyhow::Result<()> {
     utils::set_aborting_panic_hook();
-    utils::tracing::init();
 
-    let config = Config::parse();
-    let input_port = config.input_port;
+    let settings: Settings = load_settings()?;
+    ::utils::tracing::init(settings.log.rust_log.as_str(), "web-api")?;
+
+    tracing::debug!(?settings, "command-line arguments");
```
Comment (frondeus) [inline crates/api/src/main.rs:29]: I guess its no more `command-line arguments` :)

File: crates/utils/src/settings.rs [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        .to_string_lossy()
+        .to_string()
+}
+
+fn default_status_port() -> u16 {
+    3000
+}
+
+pub fn load_settings<'de, T: Deserialize<'de> + Debug>() -> anyhow::Result<T> {
+    let env = env::var("ENVIRONMENT").unwrap_or_else(|_| "development".to_string());
```
Comment (frondeus) [inline crates/utils/src/settings.rs:113]: I lack some kind of documentation (Id say in mdbook) about these environment variables.

File: crates/api/src/main.rs (hunk 1 above)
Comment (luke-biel) [reply crates/api/src/main.rs]: Changed

File: crates/utils/src/settings.rs (hunk 2 above)
Comment (luke-biel) [reply crates/utils/src/settings.rs:113]: Added index.md in docs, now it says everything about how the config is loaded

**Review by frondeus — APPROVED** [review]

**Pull request merged**